### PR TITLE
Update SDK projects to use MSBuild Sdk attribute, and hence the version of the Sdk that ships with the CLI or VS

### PR DESCRIPTION
**Pull request by dsplaisted**
In addition to cleaning our project files up and dogfooding a recent version of the SDK, this fixes the design-time build and hence intellisense in VS for me on the latest builds of VS.

### CODE REVIEW

**Comment by srivatsn** [comment]
Let's switch this to master.

**Comment by dsplaisted** [comment]
@dotnet-bot test Ubuntu14.04 Debug please

**Comment by dsplaisted** [comment]
@dotnet-bot test Windows_NT Debug please

File: src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/Microsoft.NET.Build.Tasks.UnitTests.csproj [hunk 1]
```diff
@@ -47,62 +36,42 @@
...
-    <Compile Include="GivenThatWeWantToGetDependenciesViaDesignTimeBuild.cs" />
-    <Compile Include="GivenAPublishAssembliesResolver.cs" />
-    <Compile Include="GivenADependencyContextBuilder.cs" />
-    <Compile Include="Properties\AssemblyInfo.cs" />
-    <Compile Include="Constants.cs" />
-    <Compile Include="TestLockFiles.cs" />
-  </ItemGroup>
-  <ItemGroup>
-    <None Include="all.asset.types.portable.deps.json">
+    <None Update="all.asset.types.portable.deps.json">
```
Comment (natidea) [inline src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/Microsoft.NET.Build.Tasks.UnitTests.csproj]: >Update [](start = 10, length = 6)

What does `Update` mean here?

File: test/Microsoft.NET.TestFramework/Microsoft.NET.TestFramework.csproj [hunk 2]
```diff
@@ -37,11 +26,6 @@
     </PackageReference>
```
Comment (natidea) [inline test/Microsoft.NET.TestFramework/Microsoft.NET.TestFramework.csproj]: Can we make `Version` an attribute for some of these package references?

**Review by natidea — APPROVED** [review]

**Comment by nguerrera** [comment]
This broke the signed build when I did it as part of the loc change that got reverted. Check that you're not getting default signassembly=false and do a test signed build pre merge.

**Review by nguerrera — CHANGES_REQUESTED** [review]
See above

**Comment by nguerrera** [comment]
Also check that intermediate output path remains redirected.

**Review by srivatsn — APPROVED** [review]

**Comment by srivatsn** [comment]
Once you verify the signed build, merge it in since it's an infrastructure change.

**Comment by nguerrera** [comment]
Another thing was Jenkins builds failing often due to presumed arrival of #475 -- seems nobody is seeing that anymore (and it's closed due to that). Just make sure the CLI we use has recent bits.

**Comment by nguerrera** [comment]
Discussed offline with Daniel. Closing as the same change is part of the loc change that needs to be reapplied and we don't want merge conflicts.

**Pull request closed without merging**